code-challenges.test.js

```diff
@@ -0,0 +1,105 @@
+
+const namesArrayCaps = (array) => {
+  return array.map((element) => element.name.split(" ").map((value) => value = (value[0].toUpperCase()) + value.slice(1)).join(" ") + ` is ${element.occupation}.`)
+}
+
+
+//   $ /Users/learnacademy/Desktop/week-6-assessment-jameshlee1/node_modules/.bin/jest
+//  PASS  ./code-challenges.test.js
+//   namesArrayCaps
+//     ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (1 ms)
+
+
+
+// --------------------2) Create a function that takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// a) Create a test with an expect statement using the variables provided.
+
+describe('numbersDivided', () => {
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3", () => {
+    expect(numbersDivided(hodgePodgeOne)).toEqual([ 2, 0, -1, 0 ])
+    expect(numbersDivided(hodgePodgeTwo)).toEqual([ 2, 1, -1 ])
+
+  });
+});
+
+const hodgePodgeOne = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
+// Expected output: [ 2, 0, -1, 0 ]
+const hodgePodgeTwo = [5, "Hola", 43, -34, "greetings", true]
+// Expected output: [ 2, 1, -1 ]
+
+// b) Create the function that makes the test pass.
+// Pseudo: Create a function called numbersDivided that takes in an array. Create a new variable called numbers then use the filter() method to filter the integers. Then create another variable dividedThree and use the map() built-in method to iterate through the numbers in the array and divide the number by 3. 
+
+const numbersDivided = (array) => {
+  const numbers = array.filter(element=>typeof element==='number')
+  const dividedThree = numbers.map(number=>number %3)
+  return dividedThree
+}
+
+// PASS  ./code-challenges.test.js
+//   numbersDivided
+//     ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3 (1 ms)
+
+
+// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// a) Create a test with an expect statement using the variables provided.
+
+describe('numbersArray', () => {
+  it("takes in an array of numbers and returns the sum of all the numbers cubed", () => {
+    expect(numbersArray(cubeAndSumOne)).toEqual(99)
+    expect(numbersArray(cubeAndSumTwo)).toEqual(1125)
+
+  });
+});
+
+const cubeAndSumOne = [2, 3, 4]
+// Expected output: 99
+const cubeAndSumTwo = [0, 5, 10]
+// Expected output: 1125
+
+// b) Create the function that makes the test pass.
+// Pseudo: Create a function called numbersArray that takes in an array of numbers as a parameter. Create a new variable  newNumbers then use the .map() method to take the index and cube each value. Return the variable newNumbers adding a + b
+
+const numbersArray = (array) => {
+  const newNumbers = array.map((index) => index**3)
+  return newNumbers.reduce((a, b) => a + b)
+}
```

Comment:
nice use of reduce! Your code is clean and efficient! One way to level up would be to make some of them into one-liners, but not at all necessary!